### PR TITLE
Only try to call specialized findin method for sorted input when elements are Real.

### PR DESCRIPTION
**Pull request by andreasnoack**
This should handle the issue mentioned in https://github.com/JuliaLang/julia/pull/21934#issuecomment-302873055. Eventually, we might want to handle this in a smarter way. See https://github.com/JuliaLang/julia/pull/21999.

### CODE REVIEW

File: base/array.jl [hunk 1]
```diff
@@ -1791,13 +1791,16 @@ julia> findin(a,b) # 10 is the only common element
  4
 ```
 """
-function findin(a, b)
+function findin(a::Array{<:Real}, b::Union{Array{<:Real,Real}})
```
Comment (ararslan) [inline base/array.jl]: I don't understand the union here. Did you mean `Union{Array{<:Real}, Real}`? Otherwise this is `Array{T, Real} where T<:Real`.

Comment (andreasnoack) [inline base/array.jl]: You are right. It is a typo.

**Comment by andreasnoack** [comment]
I've updated this PR with fixes to the algorithm for the issue pointed out in https://github.com/JuliaLang/julia/pull/21934#issuecomment-303033533 and added tests. I had to define `eachindex(Number)` to make this work but since `Number`s work with `getindex`, I believe this is the right thing to do anyway.

File: base/array.jl [hunk 2]
```diff
@@ -1734,6 +1734,9 @@ function _findin(a, b)
     ind
 end
 
+# If two collections are already sorted, findin can be computed with
+# a single traversal of the two collections. This is much fast than
```
Comment (tkelman) [inline base/array.jl]: faster than

**Pull request merged**